cmd/cli/options/convert.go

```diff
@@ -0,0 +1,19 @@
+package options
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// ConvertOptions defines the options for the `convert` command
+type ConvertOptions struct {
+	Format     string
+	Encoding   string
+	OutputPath string
+}
+
+// AddFlags adds command line flags for the ConvertOptions struct
+func (o *ConvertOptions) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&o.Format, "format", "f", "", "format string")
```

Comment:
We should think about default values and description that include that action supported values, I suggest pulling the list of supported formats from the library and populating some flag fields with the information.
EG: https://github.com/scribe-security/protobom/blob/main/pkg/formats/formats.go#L29

We should also consider the URI format defined my our core library.

I suggest supporting this format of a string as well go a long way with people looking at the spec, we can support other aliases as well we may find useful. (in my PR i added a URI flag you can look at).

Reply:
Yeah I know I didn't add the options to doc, but I would add them (manually)

I think supporting a URI is out of scope for this PR.

I will be adding [cyclonedx, cyclonedx-1.3, cyclonedx-1.4, spdx, spdx-2.2, spdx-2.3]

Reply:
The issue with pulling from this https://github.com/scribe-security/protobom/blob/main/pkg/formats/formats.go#L29 is that it uses the URI format, which we need translation for a more friendly alias.

Also, adding a new spec version rarely happens, thinking about cost-effectiveness.

Reply:
i actually had the same thought in my own PR and tried to tackle it already, your welcome to look in to it and decide if its useful for or not..

* Take a look at a helper function to translate between the URI list. 
[format_group.go](https://github.com/scribe-security/go-cli/blob/suggestion/basecli/internal/config/format_group.go)

* I also added to the options struct has a helper function to simply meld in the flags and spit out the URI needed to access the library. 
[application.go](https://github.com/scribe-security/go-cli/blob/suggestion/basecli/internal/config/application.go#L53)

Reply:
BY the way note that if we decide to break down the CLI in to two subcommand then the format_group can return slices instead of maps.

Reply:
dully noted. Let's open a new issue for that, sounds like a very good "first good issue" kinda thing; cool but not essential